### ISSUE TITLE
Local seed can't work when setting a chunk_size

### ISSUE THREAD

**Issue by shadowofs** [systemd/casync#77]
HI, 

This is really a work of art, and I'm gonna use it for file distribution. But recently I encountered a problem extracting .caidx file from remote. When using default chunk_size, extraction from remote server works pretty well with an exactly the same local seed. It goes like this...
```
$ bin/casync extract -v http://tosv.byted.org/obj/inf.bundle/chunk/test/1.0.0.caidx ~/test
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/1.0.0.caidx...
Seeding ./
Seeding .#197_0e590fe2_197.tar.gzfb1c07f53123bb2f
Seeded .#197_0e590fe2_197.tar.gzfb1c07f53123bb2f
Seeding 197_0e590fe2_197.tar.gz
Seeded 197_0e590fe2_197.tar.gz
Seeded ./
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/3868/3868660b7416ec4d239c8bf5a768e4a841840630d1b2f288d2b37770bab33984.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/8261/82610091911935064112266a4cb3b7665eec8fd76b024419e93933aaba9509bf.cacnk...
Extracting ./
Using feature flags: permissions
Excluding files with chattr(1) -d flag: yes
Excluding submounts: no
Digest algorithm: sha512-256
Extracting 197_0e590fe2_197.tar.gz
Extracted 197_0e590fe2_197.tar.gz
Extracted ./
Zero bytes written as sparse files: 36.7K
Bytes cloned through reflinks: 48.8M
Chunk requests fulfilled from local store: 0
Bytes used from local store: 0B
Chunk requests fulfilled from local seed: 778
Bytes used from local seed: 48.8M
Chunk requests fulfilled from remote store: 2
Bytes used from remote store: 119.9K
```

But when setting a chunk_size, like 1MB, local seed can't not work any more.
```
$ bin/casync extract -v http://tosv.byted.org/obj/inf.bundle/chunk/test/1.0.0.caidx ~/test
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/1.0.0.caidx...
Seeding ./
Seeding 197_0e590fe2_197.tar.gz
Seeded 197_0e590fe2_197.tar.gz
Seeded ./
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/f721/f7215709a07dead57703f60c370ed95ab142aeebe3e13da9501b3e0df511e302.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/947c/947c2bb717372f30705a8a59d97ef9817d8b39fb7488110fa9f76dad005833b8.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/59cb/59cb09d9269ab0bbae77663b8aefe10a93c0686a69176c9470d67e4f265c13ae.cacnk...
Extracting ./
Using feature flags: permissions
Excluding files with chattr(1) -d flag: yes
Excluding submounts: no
Digest algorithm: sha512-256
Extracting 197_0e590fe2_197.tar.gz
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/554c/554c671ff1aaabc6ab106881d0a9aca1fca7260421d109ece24b9cdaf5c19907.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/7070/70708807e28edddc04b8f25368e46de9175f196742f51b469f70eab7182e4e8b.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/4b17/4b17eb61ea11a2f2f91793d19d4021c87ed70941826a9da70826d68aa864bf7e.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/0c32/0c3267e39f69bf48a425adb2c6fae436a38fd66f82cc8789491eb8e0fdd3424b.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/fe1f/fe1f97b16407ac4699b5064ba3e171ef343333740a471cffd146e6484aaf12aa.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/931c/931c8c9c3af12464d7a74055c8851dca66aa4ce80a7114c29e74c2c76ee75886.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/4d5e/4d5e9f462f6f80d845f5adcfaaa0114d736564b694f3594a3da8c4445064c73c.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/830e/830efd015368dac241db5f60bbbdf3fc02558851c94306dbb0cf409e2b48e8b9.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/dfcf/dfcfa433ba6055bd3cbef0a04e19c1e13a43514571e9cefec810bf37cdfc172e.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/7d2e/7d2e374250997810a307f0654dd6cabb49c36bb4135471965990fd18d78497e5.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/c96c/c96c2d75785764e9c7d62e8b0562f0ee7b6d0fc848122a7ec7914a65c8b75cb4.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/23f2/23f219787e9fa775d34ef71ddc067c18ebf2ae12d52a4434080713d6699b403f.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/5e26/5e267b676c3171ddb27e3204beb77090aab7b3d9c66f1aabf55f16fd652df71f.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/a178/a1788c659e6f0dc8540b0b44c3613fd2b61302c4e63b9987ec59f8b391326917.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/4b68/4b680c8db35fff65bdb1995e0f3f96f007552f4b0d42aca78466b9d52b4b80c2.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/1083/10830d6184eb96875cf78936e6f3be03eb04796ae5dd340788b58ccecca3751d.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/075d/075dcd35dcc3c3ca937b071941287c6aef9d861fc1076db94aa0cbb8ac440d89.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/b0ec/b0ec9fc9feb97b97f2434b23beec3bf9b9158d28769289adafdccd89611b3c72.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/ac11/ac110a3cebb08288666995fe3061ec2a52927c501c4e158bbb4d61660ea416b7.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/9b01/9b0183f59bdc15348ed8ddffd9729cd7d8742d47f6c879dc1820d13261981236.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/8ab9/8ab98012f0259dda789f7076e36e7aa978c08fcdb10215ad50d5ce15fc98633b.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/c0fe/c0fe1ed5015d2a2b2fb1f16e0dcf3d7a18b18ba361fdd1c7cab4ba728f6a7100.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/5e7b/5e7bf1f903ee565c6109eb97fc5fa02b72133fa0d6a3c0d43af57e1590f39b9f.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/c804/c8046fe39f08197f3966546d45d41cb267eec0bde61aed149744d4417f7f9a00.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/042f/042f70bb4b86486bda15e0309cfd880b664d5f5b964d49d0714bd23e10256e07.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/25a9/25a9ccaee30520d517b0b48cc408d6f70147e7d4bfab1cdbc36e9528cedc2ea1.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/5bc4/5bc490c8196f4bdce35313cfa84ed0562d5a5a0e398ed63f7f632d71e49150e2.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/cbd6/cbd6b5ba8efa943e6593212ec2766afba02af94c2fadfff2b5dc5039fd42dc96.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/85f1/85f11aa43807dfc43b6d863002ee7fb72c216cb9bc2e65d3c96ed07d4a35b888.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/33d0/33d0c166f174321629760a8736267bc3f3f785f61bf9fc0049c2cde0edd5fed0.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/729a/729adcec46361a2522be3d7ba1e05dc52d0f357ef7d42830f2f8b08bfcb725dd.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/0c8f/0c8f04465c002e908f86b6583aff9554ba9ce116902bda553044550e8b65c5b0.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/1ae9/1ae902009b5c0428769f9321f272fd6e6928c3cfa375b5a28f85e4e7d34c0c02.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/d8d8/d8d824e454196d8ac4fe8b81163a696caa8163e831f41a647745ec07ad3ea24c.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/e3f8/e3f84f563825c5f8b8b19f917f6b7bc44d692b5bf866193f5339600a3246365b.cacnk...
Acquiring http://tosv.byted.org/obj/inf.bundle/chunk/test/default.castr/8936/89366ea0edbea5ffcbfa5f08ca8b5fa8edf3de3977556ae6918a3b44b3d0712f.cacnk...
Extracted 197_0e590fe2_197.tar.gz
Extracted ./
Zero bytes written as sparse files: 36.7K
Bytes cloned through reflinks: 0B
Chunk requests fulfilled from local store: 0
Bytes used from local store: 0B
Chunk requests fulfilled from local seed: 0
Bytes used from local seed: 0B
Chunk requests fulfilled from remote store: 39
Bytes used from remote store: 48.9M
```

Test version: v2. 
local lib versions:
```
$ ldd bin/casync
        linux-vdso.so.1 (0x00007ffdbccab000)
        libacl.so.1 => /lib/x86_64-linux-gnu/libacl.so.1 (0x00007f18ef079000)
        libfuse.so.2 => /lib/x86_64-linux-gnu/libfuse.so.2 (0x00007f18eee3b000)
        liblzma.so.5 => /lib/x86_64-linux-gnu/liblzma.so.5 (0x00007f18eec15000)
        libselinux.so.1 => /lib/x86_64-linux-gnu/libselinux.so.1 (0x00007f18ee9f0000)
        libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f18ee7d5000)
        libzstd.so.1 => /usr/lib/x86_64-linux-gnu/libzstd.so.1 (0x00007f18ee550000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f18ee24f000)
        libcrypto.so.1.0.0 => /usr/lib/x86_64-linux-gnu/libcrypto.so.1.0.0 (0x00007f18ede53000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f18edc36000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f18ed88b000)
        libattr.so.1 => /lib/x86_64-linux-gnu/libattr.so.1 (0x00007f18ed686000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f18ed482000)
        libpcre.so.3 => /lib/x86_64-linux-gnu/libpcre.so.3 (0x00007f18ed214000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f18ef282000)
```

There's definitely a bug here. I'm trying to solve this myself, but it seems a bit difficult for me. :(
Thanks for any help.

**Comment by shadowofs** [comment]
Fixed. 
https://github.com/systemd/casync/pull/78

**Comment by poettering** [comment]
Closing, because fixed in #78
